upload/admin/controller/module/smaily_for_opencart.php

```diff
@@ -141,7 +147,37 @@ public function index() {
         // Rss feed.
         $data['rss_feed_title'] = $this->language->get('rss_feed_title');
         $data['rss_feed_text']  = $this->language->get('rss_feed_text');
-
+        $data['smaily_rss_url_base'] = $this->config->get('config_url') . 'index.php?route=smailyforopencart/rss';
+        $query_parameters = array();
+        if ($this->config->get('smaily_for_opencart_rss_category')) {
+            $query_parameters['category'] = $this->config->get('smaily_for_opencart_rss_category');
+        }
+        if ($this->config->get('smaily_for_opencart_rss_sort_by')) {
+            $query_parameters['sort_by'] = $this->config->get('smaily_for_opencart_rss_sort_by');
+        }
+        if ($this->config->get('smaily_for_opencart_rss_sort_order')) {
+            $query_parameters['sort_order'] = $this->config->get('smaily_for_opencart_rss_sort_order');
+        }
+        if ($this->config->get('smaily_for_opencart_rss_limit')) {
+            $query_parameters['limit'] = $this->config->get('smaily_for_opencart_rss_limit');
+        }
+        $data['smaily_rss_url'] = $this->config->get('config_url') . 'index.php?route=smailyforopencart/rss&'
+            . http_build_query($query_parameters);
```

Comment:
You could D.R.Y-up RSS URL compilation:

```suggestion
        $data['smaily_rss_url'] = $data['smaily_rss_url_base'] . '&' . http_build_query($query_parameters);
```

Reply:
Yes, good idea, added